storage-node/src/storage_manager.rs

```diff
@@ -20,32 +22,48 @@ use crate::{
 pub struct StorageManager<C: ParpulseCache> {
     // TODO: Consider making the cache lock-free.
     cache: RwLock<C>,
-    disk_manager: Mutex<DiskManager>,
+    disk_manager: Mutex<DiskManagerSync>,
     // TODO: cache_base_path should be from config
     // Cache_key = S3_PATH; Cache_value = (CACHE_BASE_PATH + S3_PATH, size)
     cache_base_path: String,
 }
 
 impl<C: ParpulseCache> StorageManager<C> {
-    pub fn new(cache: C, disk_manager: DiskManager, cache_base_path: String) -> Self {
+    pub fn new(cache: C, disk_manager: DiskManagerSync, cache_base_path: String) -> Self {
         Self {
             cache: RwLock::new(cache),
             disk_manager: Mutex::new(disk_manager),
             cache_base_path,
         }
     }
 
-    pub async fn get_data(&self, request: RequestParams) -> ParpulseResult<usize> {
-        // 1. Try to get data from the cache first.
-        // 2. If cache miss, then go to storage reader to fetch the data from
-        // the underlying storage.
-        // 3. If needed, update the cache with the data fetched from the storage reader.
+    // TODO: This is a dummy function.
+    fn parse_s3_request(&self, request: &str) -> ParpulseResult<(String, Vec<String>)> {
+        let bucket = "tests/parquet/".to_string();
```

Comment:
better abide by the S3 bucket naming convention
```suggestion
        let bucket = "tests-parquet".to_string();
```